src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/TransferApprovalStatus/TransferApprovalCommandHandler.cs

```diff
@@ -57,6 +77,31 @@ public TransferApprovalCommandHandler(IEmployerCommitmentApi commitmentsApi)
             {
                 await _commitmentsService.PatchTransferApprovalStatus(message.TransferSenderId, message.CommitmentId, request);
             }
+
+            await SendNotifications(commitment, message.TransferStatus);
+        }
+
+        private async Task SendNotifications(CommitmentView commitment, Commitments.Api.Types.TransferApprovalStatus newTransferApprovalStatus)
+        {
+            if (!_configuration.CommitmentNotification.SendEmail)
+            {
+                _logger.Info("Sending email notifications disabled by config.");
+                return;
+            }
+
+            var tokens = new Dictionary<string, string>
```

Comment:
Would it be better (consistent) if all tokenising was done inside the ProviderEmailNotificationService methods?

Reply:
When I wrote it, I was in two minds between tokenising in each EmailNotificationService (for consistency), or only generating the tokens once, but I think you're right and centralising all token generation into [Provider|Employer]EmailNotificationService is more important than a micro-optimisation.
Token generation duly moved! :smiley: